sled-storage/src/manager.rs

```diff
@@ -526,7 +524,7 @@ impl StorageHandle {
     // and ask for the set of all datasets from Nexus.
     pub async fn upsert_filesystem(
```

Comment:
FWIW, this is the **only** call to `upsert_filesystem`. Once we remove the invocation in `omicron_zones_ensure`, we can remove this function (and this "optional dataset UUID" argument) altogether.